proxy/http/HttpSM.cc

```diff
@@ -1954,9 +1954,7 @@ HttpSM::state_read_server_response_header(int event, void *data)
     t_state.api_next_action       = HttpTransact::SM_ACTION_API_READ_RESPONSE_HDR;
 
     // if exceeded limit deallocate postdata buffers and disable redirection
-    if (enable_redirection && (redirection_tries < t_state.txn_conf->number_of_redirections)) {
-      ++redirection_tries;
-    } else {
+    if (enable_redirection && (redirection_tries >= t_state.txn_conf->number_of_redirections)) {
```

Comment:
Yes, I asked about this because I was not sure whether we should be checking the response here. The problem that I see with this change is that the condition disagrees with ``is_redirect_required()`` when ``redirection_tries`` is equal to ``number_of_redirections``.